### ISSUE TITLE
On a fresh install, opening download data dialogue produces stack trace

### ISSUE THREAD

**Issue by hsimonoko** [faunalia/qgreenland-plugin#32]
I just installed the plugin, and I clicked Web -> Qgreenland -> Download Data and it produced a stack trace: 
```
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str' 
Traceback (most recent call last):
  File "/Users/hosi8088/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/qgreenland-plugin/qgreenland.py", line 207, in run_download
    self.dlg = QGreenlandDownload()
  File "/Users/hosi8088/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/qgreenland-plugin/qgreenland_dowload.py", line 128, in __init__
    self._fill_tree()
  File "/Users/hosi8088/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/qgreenland-plugin/qgreenland_dowload.py", line 311, in _fill_tree
    url = QUrl(self.downloading_url + 'manifest.json')
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'

Python version: 3.8.7 (default, Feb 10 2021, 09:04:08) [Clang 12.0.0 (clang-1200.0.32.29)] 
QGIS version: 3.16.11-Hannover Hannover, 26cc1c762f 

Python Path:
/Applications/QGIS-LTR.app/Contents/MacOS/../Resources/python
/Users/hosi8088/Library/Application Support/QGIS/QGIS3/profiles/default/python
/Users/hosi8088/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
/Applications/QGIS-LTR.app/Contents/MacOS/../Resources/python/plugins
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/rasterio-1.1.5-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/Fiona-1.8.13.post1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/numpy-1.20.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/patsy-0.5.1-py3.8.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/matplotlib-3.3.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python38.zip
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/click_plugins-1.1.1-py3.8.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/affine-2.3.0-py3.8.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/pandas-1.1.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/opencv_contrib_python-4.3.0.36-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/statsmodels-0.11.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/netCDF4-1.5.4-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/pyproj-2.6.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/GDAL-3.2.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/scipy-1.5.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/geopandas-0.8.1-py3.8.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/Rtree-0.9.4-py3.8.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/snuggs-1.4.7-py3.8.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/cftime-1.2.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/site-packages/Pillow-7.2.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS-LTR.app/Contents/MacOS/lib/python3.8/lib-dynload
/Users/hosi8088/Library/Application Support/QGIS/QGIS3/profiles/default/python
```

**Comment by MattF-NSIDC** [comment]
@ghtmtt can you please add Hope as a maintainer so she can add labels to issues? Thanks!

**Comment by hsimonoko** [comment]
When I try Web -> Qgreenland -> Download Data again, I get this error: 
```
2021-12-14T10:14:32     WARNING    Traceback (most recent call last):
              File "/Users/hosi8088/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/qgreenland-plugin/qgreenland.py", line 210, in run_download
              self.dlg.show()
             AttributeError: 'QGreenland' object has no attribute 'dlg'
```

**Comment by ghtmtt** [comment]
the second error is related do the first one (when the plugin fails it also fails to create the instance of the `QGreenland` class. So if you restart QGIS this error should not appear. 

The first error is weird. If you didn't already choose the server url then by clicking on the `Download data` menu a dialog should appear informing you that you haven't any url set up yet and the server dialog should pop up: https://github.com/faunalia/qgreenland-plugin/blob/master/qgreenland_dowload.py#L161 @MattF-NSIDC @trey-stafford did you ever get this error?

**Comment by MattF-NSIDC** [comment]
@ghtmtt When @hsimonoko first installed the plugin, we navigated to the `Config` dialog first, clicked OK, then opened the `Download data` dialog second, and experienced the error. We were able to access the `Config` dialog with no issues.

**Comment by ghtmtt** [comment]
fixed with #34
